Zotero/Scenes/Detail/SingleCitation/Views/SingleCitationViewController.swift

```diff
@@ -46,95 +46,144 @@ final class SingleCitationViewController: UIViewController {
-        self.navigationItem.rightBarButtonItem?.isEnabled = !state.loadingPreview
-
-        if let error = state.error {
-            switch error {
-            case .styleMissing:
-                self.coordinatorDelegate?.showMissingStyleError()
-
-            case .cantPreloadWebView:
-                self.coordinatorDelegate?.showCitationPreview(errorMessage: L10n.Errors.citationPreview)
+            default:
+                break
             }
+
+            previewWebView.scrollView.isScrollEnabled = false
+            previewWebView.backgroundColor = .clear
+            previewWebView.scrollView.backgroundColor = .clear
+            previewWebView.configuration.userContentController.add(self, name: "heightHandler")
         }
 
-        if state.changes.contains(.preview) {
-//            do {
-//                if let data = state.preview.data(using: .utf8) {
-//                    let attr = try NSAttributedString(data: data, options: [.documentType: NSAttributedString.DocumentType.rtf, .characterEncoding: String.Encoding.utf8], documentAttributes: nil)
-//                    self.previewTitleLabel.numberOfLines = 0
-//                    self.previewTitleLabel.attributedText = attr
-//                }
-//            } catch let error {
-//                NSLog("ERROR: \(error)")
-//            }
-            self.updatePreferredContentSize()
+        func setupNavigationBar() {
+            let cancel = UIBarButtonItem(title: L10n.cancel, style: .plain, target: nil, action: nil)
+            cancel.rx.tap.subscribe(onNext: { [weak self] in
+                self?.navigationController?.presentingViewController?.dismiss(animated: true, completion: nil)
+            })
+            .disposed(by: disposeBag)
+            navigationItem.leftBarButtonItem = cancel
+
+            setupRightButtonItem(isLoading: false)
+            navigationItem.rightBarButtonItem?.isEnabled = false
         }
 
-        if state.changes.contains(.copied) {
-            self.navigationController?.presentingViewController?.dismiss(animated: true, completion: nil)
+        func setupObserving() {
+            viewModel.stateObservable
+                .subscribe(onNext: { state in
+                    update(state: state)
+                })
+                .disposed(by: disposeBag)
+
+            locatorTextField.rx.controlEvent(.editingChanged).flatMap({ Observable.just(self.locatorTextField.text ?? "") })
+                .debounce(.milliseconds(150), scheduler: MainScheduler.instance)
+                .subscribe(onNext: { [weak self] value in
+                    self?.viewModel.process(action: .setLocatorValue(value))
+                })
+                .disposed(by: disposeBag)
+
+            omitAuthorSwitch.rx.controlEvent(.valueChanged)
+                .subscribe(onNext: { [weak self] _ in
+                    guard let self else { return }
+                    viewModel.process(action: .setOmitAuthor(omitAuthorSwitch.isOn))
```

Comment:
Doesn't this need `self.viewModel.process(...)`? Or is `self` automatically added here too?